src/core/batch.rkt

```diff
@@ -31,6 +38,57 @@
     [(list op args ...) (cons op (map f args))]
     [_ expr]))
 
+(define (batchref-all-subnodes x #:reverse? [reverse? #f])
+  ;; No recursion on approx-spec
```

Comment:
It is ugly to have two of these; I think the ultimate solution is we need to stop putting specs in batches anywhere.